jlm/llvm/backend/jlm2llvm/instruction.cpp

```diff
@@ -324,18 +324,45 @@ convert(
       ctx);
 }
 
+static void
+CreateStoreInstruction(
+    const variable * address,
+    const variable * value,
+    bool isVolatile,
+    size_t alignment,
+    ::llvm::IRBuilder<> & builder,
+    context & ctx)
+{
+  auto storeInstruction = builder.CreateStore(ctx.value(value), ctx.value(address), isVolatile);
+  storeInstruction->setAlignment(::llvm::Align(alignment));
+}
+
 static inline ::llvm::Value *
 convert_store(
-    const rvsdg::simple_op & op,
-    const std::vector<const variable *> & args,
+    const rvsdg::simple_op & operation,
+    const std::vector<const variable *> & operands,
     ::llvm::IRBuilder<> & builder,
     context & ctx)
 {
-  JLM_ASSERT(is<StoreOperation>(op) && args.size() >= 2);
-  auto store = static_cast<const StoreOperation *>(&op);
+  auto storeOperation = util::AssertedCast<const StoreOperation>(&operation);
+  CreateStoreInstruction(
+      operands[0],
+      operands[1],
+      false,
+      storeOperation->GetAlignment(),
+      builder,
+      ctx);
+  return nullptr;
+}
 
-  auto i = builder.CreateStore(ctx.value(args[1]), ctx.value(args[0]));
-  i->setAlignment(::llvm::Align(store->GetAlignment()));
+static ::llvm::Value *
+convert(
```

Comment:
I assume there is a historic reason why the above function is called `convert_store` and takes an untyped `simple_op`, while this one doesn't?

Reply:
Yes, some cleanup and refactoring is necessary in this pass. However, that is for another PR.